app/controllers/floating_ip_controller.rb

```diff
@@ -25,20 +25,14 @@ def button
     @refresh_div = "main_div"
 
     case params[:pressed]
-    when "floating_ip_tag"
-      tag("FloatingIp")
     when 'floating_ip_delete'
       delete_floating_ips
     when "floating_ip_edit"
       javascript_redirect(:action => "edit", :id => checked_item_id(params))
     when "floating_ip_new"
       javascript_redirect(:action => "new")
     else
-      if !flash_errors? && @refresh_div == "main_div" && @lastaction == "show_list"
-        replace_gtl_main_div
-      else
-        render_flash
-      end
+      super
```

Comment:
The code is a little tangled, where this `super` ends up?

Reply:
`super` calls `button` method [here](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/mixins/generic_button_mixin.rb#L43).

Reply:
Okay, but as far as I see the method above does something different, sets some params. Did you go through all of the possible side effects?

Reply:
Yes, the pattern is far from great, but it's not the best we have and we are unifying it into this form (before, later we can so something better).

Reply:
From my experience, `button` in _GenericButtonMixin_ works much better than any `button` method defined in any controller (some other controllers' `button` methods need refactoring, too). It cares about more scenarios, possibilities, rendering flash messages etc. Also later if we decide to add other buttons for already existing actions (like removing etc.) to toolbar in Floating Ip List View/Summary screen and related screens (lists of items displayed through Floating Ip Summary), the actions will work very probably without any other changes in FloatingIp controller.

Reply:
> From my experience, `button` in _GenericButtonMixin_ works much better than any `button` method defined in any controller. It cares about more scenarios, possibilities, rendering flash messages etc.

Thank you ;-)

I am aware of the fact that the method is ugly, but it's less ugly than the other ones. And if we make the controllers use the same method or the same pattern thatn we have less trouble moving forward towards a better button press handling code in the future.

Reply:
super :laughing: